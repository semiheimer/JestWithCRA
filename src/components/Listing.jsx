import React from 'react';
import {useFetch} from '../hooks/useFetch';

const Listing = () => {
    const {data, isLoading} = useFetch();

    if (isLoading) return <div>Loading...</div>;
    return (
        <ul>
            {data.map(post => (
                <li key={post.id}>{post.title}</li>
            ))}
        </ul>
    );
};

export default Listing;
