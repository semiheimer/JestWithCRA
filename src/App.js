import React from 'react';
import Listing from './components/Listing';
import {di} from 'react-magnetic-di';

function App() {
    di(<Listing />);

    return (
        <div>
            <div>Hello</div>
            <Listing />
            <div>Bello</div>
        </div>
    );
}

export default App;
