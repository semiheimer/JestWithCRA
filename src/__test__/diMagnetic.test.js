import {render} from '@testing-library/react';
import App from '../App';
import {injectable, DiProvider} from 'react-magnetic-di';
import Listing from '../components/Listing';

const mockListingComponent = () => <div data-testid1="listing" />;
const ListingDi = injectable(Listing, () => <div data-testid="listing" />);

describe('App', () => {
    it('should fetch Listing component with magnetic', async () => {
        const {container} = render(<App />, {
            wrapper: props => <DiProvider use={[ListingDi]} {...props} />,
        });
        expect(container).toMatchSnapshot();
    });
});
