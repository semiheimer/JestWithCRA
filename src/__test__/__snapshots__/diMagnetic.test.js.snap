// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should fetch Listing component with magnetic 1`] = `
<div>
  <div>
    <div>
      Hello
    </div>
    <div>
      Loading...
    </div>
    <div>
      Bello
    </div>
  </div>
</div>
`;
